### ISSUE TITLE
找不到定义

### ISSUE THREAD

**Issue by Tbetter** [ZZYG0g0g0/AC-XWCoDe#1]
对您能公开代码表示感激！！
我阅读了您的论文和代码。
在我想要运行时，`xgboost_classification_test_threshold3`找不到定义。
是否可以请教一下，哪里出现了问题？

**Comment by ZZYG0g0g0** [comment]
你好，可以把问题具体一点吗，比如在哪个文件的哪一行出的问题？我没有找到你出现问题的地方，不好意思。

**Comment by Tbetter** [comment]
<img width="1506" alt="截屏2024-12-18 15 09 10" src="https://github.com/user-attachments/assets/04f4b5b1-82d3-4555-832c-62c105578d4f" />
在这个app.py当中，第75行。

**Comment by ZZYG0g0g0** [comment]
很抱歉，我来得及对提交的代码进行修改，所提交的代码还存在部分漏洞，事实上，上述代码是该论文的一个会议版本的前置工作中的代码，你可以在我的github页面中的XWCoDe项目中找到，然而我忘记删除了，本论文中的代码主要在evaluation/RQ1中，如需运行，则运行该代码即可，你所提出的问题包含的代码片段为
def xgboost_classification_test_threshold3(test_size, dataset_name, feature_file_name):
    df = pd.read_excel('../../datasets/'+dataset_name+'/'+feature_file_name+'.xlsx')
    labels = pd.read_excel('../../datasets/'+dataset_name+'/labels.xlsx')

    # 数据归一化
    scaler = MinMaxScaler()
    X_normalized = scaler.fit_transform(df)

    results_by_test_size = []

    for t_size in test_size:
        threshold3 = [i/100 for i in range(1, 101)]
        # 生成数据的索引
        indices = np.arange(len(X_normalized))
        X_train, X_test, y_train, y_test, train_index, test_index = train_test_split(X_normalized, labels, indices, test_size=t_size, random_state=42)
        clf = xgb.XGBClassifier(use_label_encoder=False, eval_metric='logloss')
        clf.fit(X_train, y_train.values.ravel())
        y_pred = clf.predict_proba(X_test)[:, 1]
        temp_results = []
        for t in threshold3:
            y_pred_fb = (y_pred >= t).astype(int)
            # 直接计算精确率和召回率
            precision = precision_score(y_test, y_pred_fb)
            recall = recall_score(y_test, y_pred_fb)
            f1 = 2 * (precision * recall) / (precision + recall) if (precision + recall) > 0 else 0
            temp_results.append((t, precision, recall, f1))#加一个元组
        results_by_test_size.append((t_size, temp_results))
    return results_by_test_size

**Comment by Tbetter** [comment]
谢谢，感谢您的解答
